include/nwutil.h

```diff
@@ -32,6 +33,30 @@ unsigned nwutil_http_proxy_port(nwutil_http_proxy_settings_t *settings);
 /* Deallocate the data structure holding the HTTP proxy settings. */
 void nwutil_release_http_proxy_settings(nwutil_http_proxy_settings_t *settings);
 
+typedef struct nwutil_url nwutil_url_t;
+
+/*
+ * Implementation of the basic URL parser from the WHATWG URL
+ * Standard. Limitations:
+ *
+ * - no support for the encoding override, url and state override
+ *   arguments
+ *
+ * - missing IDNA to ASCII validation and encoding of the host string
+ */
+nwutil_url_t *nwutil_parse_url(const char *buffer,
+                               size_t size,
```

Comment:
It would be worth mentioning explicitly what—if any—role a NUL character in buffer might have.